### PR TITLE
Make `MultiSigner` use compressed ECDSA public key

### PR DESCRIPTION
**Pull request by bkchr**
`MultiSigner` did not compressed the ECDSA public key before hashing it but `MultiSignature` expected that the public key is compressed when verifying a signature. Thus, the signature verification could never succeed. 

This pr changes the following:
- `MultiSigner` compresses the public key before hashing it with `blake256`.
- `Verify for ecdsa::Signature` compresses given public key before comparing it.
- `ecdsa::Public` is now an enum that distinguishes between the full and compressed format.

Fixes: #4498

### CODE REVIEW

File: primitives/runtime/src/lib.rs [hunk 1]
```diff
@@ -246,7 +246,9 @@ impl traits::IdentifyAccount for MultiSigner {
 		match self {
 			MultiSigner::Ed25519(who) => <[u8; 32]>::from(who).into(),
 			MultiSigner::Sr25519(who) => <[u8; 32]>::from(who).into(),
-			MultiSigner::Ecdsa(who) => sp_io::hashing::blake2_256(who.as_ref()).into(),
+			MultiSigner::Ecdsa(who) => sp_io::hashing::blake2_256(
+				&who.as_compressed().expect("what should we do?")[..],
```
Comment (bkchr) [inline primitives/runtime/src/lib.rs]: @gavofyork what should we do here? I think the best would be to propagate the error via `IdentifyAccount`.

Comment (gavofyork) [reply primitives/runtime/src/lib.rs]: i believe this cannot fail for any valid keypair, so `.expect` should be fine

**Comment by gavofyork** [comment]
but... what's the point of supporting uncompressed public key format?

**Comment by bkchr** [comment]
> but... what's the point of supporting uncompressed public key format?

Good question, I just followed the initial implementation. However, I think that for ergonomic reasons the support for both sounds reasonable to me. If you say that people only use the compressed version, I can remove the support for full public keys.

**Comment by gavofyork** [comment]
it does leave this annoying `.expect` thing in.

**Pull request merged**